src/App.tsx

```diff
@@ -15,47 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+function getPreparedGoods(
+  goods: string[], sortBy: SortType, reverse: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortBy === SortType.alphabet) {
+    preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
+
+  if (sortBy === SortType.length) {
+    preparedGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (reverse) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortType>(SortType.default);
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortBy, reversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button', 'is-info', { 'is-light': sortBy !== SortType.alphabet },
+          )}
+          onClick={() => setSortBy(SortType.alphabet)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button', 'is-success', { 'is-light': sortBy !== SortType.length },
+          )}
+          onClick={() => setSortBy(SortType.length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button', 'is-warning', { 'is-light': !reversed },
+          )}
+          onClick={() => setReversed(!reversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortBy || reversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortBy(SortType.default);
+              setReversed(false);
```

Comment:
It's better to move this code to the separate function, because it's a bad practice to write a lot of code in a template